nautobot/docs/development/core/index.md

```diff
@@ -69,6 +69,30 @@ Nautobot adheres to the Semantic Versioning ("SemVer") strategy, which gives us
 
 For more information, please visit [SemVer.org](https://semver.org/).
 
+#### Public Interface
+
+The guiding principals behind Nautobot's SemVer strategy is scoped to adhering to the prescribed public interface. This includes both the application operators and application developer. Anything not explicitly covered here is presumed to not be covered under SemVer.
+
+* The REST API interface within a Version, will not change, but may add to
+    * URL structures
+    * Expected request and response formats
+    * Behavior of each endpoint
+    * Status and error codes
+    * The HTTP verbs
+* Graphql data structures
+* URL structures for primary navigation and feature access
+* Any Django model and their relationships for models provided by Nautobot
+* Functions, classes, and modules in `nautobot.apps.*` and their associated signature
+* Any management commands provided by Nautobot
+* Documented configuration options
+
+While anything not explicitly stated above should be not be considered scoped to SemVer, the following should be strictly avoided.
+
+* Functions or methods that start with an underscore (e.g., `_private_func`)
+* Any functions, classes, and modules **not** in `nautobot.apps.*`.
+* Any modules, classes, functions, or feature that are explicitly marked as "internal" or "experimental" in the documentation
+* Any features or functionalities that aren't part of the official documentation should not be relied upon.
```

Comment:
This bullet list, at least, still feels like it belongs in the `apps` and `jobs` development documentation rather than here. Maybe at a minimum those docs should cross-link to this section?